src/internal/m365/collection/site/collection.go

```diff
@@ -171,7 +151,13 @@ func (sc *Collection) streamLists(
 		wg      sync.WaitGroup
 	)
 
-	defer sc.finishPopulation(ctx, &metrics)
+	defer finishPopulation(
+		ctx,
+		sc.stream,
+		sc.statusUpdater,
+		sc.fullPath,
+		metrics,
+	)
```

Comment:
I can't comment on unedited lines. But one comment about the `semaphoreCh`  on line 166. Although it has a size of 5, we are serializing all requests i.e. we call `sc.handleListItems` for one `listID` at a time. 

Maybe it's missing a `go` keyword in front? It could be handled in a separate PR, just wanted to bring it up.

Reply:
`go` keyword infornt of `sc.handleListItems...`?
you suspect if there are, say 1000 lists, it will block the `semaphoreCh` because we have a smaller channel siz of `5`?

Reply:
> go keyword infornt of sc.handleListItems...?

yes.

> you suspect if there are, say 1000 lists, it will block the semaphoreCh because we have a smaller channel siz of 5?

No. Currently we are downloading 1 list item at a time with prefetch collection. Using a goroutine will help us do 5 at a time since that's the semaphore capacity.

Do note that once we switch to lazy fetch collection for lists, `semaphoreCh` will only be useful for parallelizing list item enumeration. It will no longer be relevant for item data download since that's deferred to lazy reader i.e. parallelism for  list **data** download will be dictated by kopia.